src/sclient/StorageClient.cpp

```diff
@@ -0,0 +1,235 @@
+    if (!partsAllocResult.ok()) {
+        LOG(ERROR) << "Get parts alloc for space: " << spaceName
+                   << " failed: " << partsAllocResult.status();
+        return result;
+    }
+    auto partsAlloc = partsAllocResult.value();
+    for (auto &partAlloc : partsAlloc) {
+        auto partId = partAlloc.first;
+        auto leaderResult = sclient_->getLeader(spaceId, partId);
+        if (!leaderResult.ok()) {
+            LOG(ERROR) << "Get leader host for space: " << spaceName << ", part: " << partId
+                       << " failed: " << leaderResult.status();
+            return result;
+        }
+        result[partId] = leaderResult.value();
+    }
+    DCHECK(result.size() == partsAlloc.size());
+
+    return result;
+}
+
+ScanEdgeIter StorageClient::scanEdgeWithPart(std::string spaceName,
+                                             int32_t partId,
+                                             std::string edgeName,
+                                             std::vector<std::string> propNames,
+                                             int64_t limit,
+                                             int64_t startTime,
+                                             int64_t endTime,
+                                             std::string filter,
+                                             bool onlyLatestVersion,
+                                             bool enableReadFromFollower) {
+    auto spaceIdResult = mclient_->getSpaceIdByNameFromCache(spaceName);
+    if (!spaceIdResult.ok()) {
+        LOG(ERROR) << "Get space id for " << spaceName << " failed: " << spaceIdResult.status();
+        return ScanEdgeIter(nullptr, nullptr, false);
+    }
+    int32_t spaceId = spaceIdResult.value();
+    auto edgeTypeResult = mclient_->getEdgeTypeByNameFromCache(spaceId, edgeName);
+    if (!edgeTypeResult.ok()) {
+        LOG(ERROR) << "Get edge type for " << edgeName << " failed: " << edgeTypeResult.status();
+        return ScanEdgeIter(nullptr, nullptr, false);
+    }
+    int32_t edgeType = edgeTypeResult.value();
+
+    storage::cpp2::EdgeProp returnCols;
+    returnCols.set_type(edgeType);
+    returnCols.set_props(propNames);
+
+    auto *req = new storage::cpp2::ScanEdgeRequest;
+    req->set_space_id(spaceId);
+    req->set_part_id(partId);
+    req->set_cursor("");
+    req->set_return_columns(returnCols);
+    req->set_limit(limit);
+    req->set_start_time(startTime);
+    req->set_end_time(endTime);
+    req->set_filter(filter);
+    req->set_only_latest_version(onlyLatestVersion);
+    req->set_enable_read_from_follower(enableReadFromFollower);
+
+    return ScanEdgeIter(sclient_.get(), req);
+}
+
+ScanVertexIter StorageClient::scanVertexWithPart(std::string spaceName,
+                                                 int32_t partId,
+                                                 std::string tagName,
+                                                 std::vector<std::string> propNames,
```

Comment:
the keys of the vertex is unknown by users, you need to add the key prop of the edge.

```
'_vid'
```